linetools/isgm/lls.py

```diff
@@ -0,0 +1,540 @@
+        pix_kludge = np.where((wv_rest > 911.5*u.AA) & (wv_rest < 912.8*u.AA))[0]
+        tau_model[pix_kludge] = tau_model[pix_LL]
+        
+        # Fill in flux
+        model = copy.deepcopy(spec)
+        model.flux = np.exp(-1. * tau_model).value
+
+        # Smooth?
+        if smooth > 0:
+            model.gauss_smooth(smooth)
+
+        # Return
+        return model
+
+    def get_zpeak(self):
+        """ Measure zpeak from an ionic transition
+        """
+        if self.ions is None:
+            print('get_zpeak: Need to fill ions with get_ions first.')
+            return
+
+        # Ions for analysis
+        low_ions = [ (14,2), (6,2), (13,2), (26,2), (13,3)]
+        high_ions= [(14,4), (6,4)]
+
+        for tt in range(4):
+            if tt == 0:
+                ions = low_ions
+                iflg = 1 # Standard
+            elif tt == 1:
+                ions = low_ions
+                iflg = 2 # Saturated
+            elif tt == 2:
+                ions = high_ions
+                iflg = 1 # Standard
+            elif tt == 3:
+                ions = high_ions
+                iflg = 2 # Standard
+            else:
+                raise ValueError('Bad value')
+
+            # Search 
+            for ion in ions:
+                try:
+                    t = self.ions[ion]
+                except KeyError:
+                    continue
+                # Measurement?
+                if t['flg_clm'] == iflg:
+                # Identify the transition
+                    gdi = np.where( (self.ions.trans['Z'] == ion[0]) &
+                                (self.ions.trans['ion'] == ion[1]) &
+                                (self.ions.trans['flg_clm'] <= iflg) )[0]
+                    # Take the first one
+                    gdt = self.ions.trans[gdi[0]]
+                    wrest = gdt['wrest']
+                    flgs = self.clm_analy.clm_lines[wrest].analy['FLAGS']
+                    spec_file = self.clm_analy.fits_files[flgs[1] % 64]
+                    # Generate an Abs_Line with spectrum
+                    line = AbsLine(wrest, z=self.clm_analy.zsys, spec_file=spec_file)
+                    # vpeak
+                    from linetools import utils as ltu
+                    vpeak = line.vpeak()
+                    self.zpeak = ltu.z_from_v(self.clm_analy.zsys, vpeak)
+                    if tt == 3:
+                        print('zpeak WARNING: Using saturated high-ions!!')
+                    break
+            else:
+                continue
+            break
+
+        # Error catching
+        if self.zpeak is None:
+            # Skip primordial LLS
+            print('lls.zpeak: No transition in {:s}'.format(self.clm_analy.clm_fil))
+            return (0,0), 0.
+        # Return
+        return ion, vpeak
+
+    # Output
+    def __repr__(self):
+        return ('[{:s}: {:s} {:s}, zabs={:g}, NHI={:g}, tau_LL={:g}, [M/H]={:g} dex]'.format(
+                self.__class__.__name__,
```

Comment:
I prefer [Z/H] to [M/H] (because of X, Y, Z notation for metals), but not a big deal.